plugins/aws/resoto_plugin_aws/resource/cloudtrail.py

```diff
@@ -0,0 +1,226 @@
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def collect_trail(trail_arn: str) -> None:
+            if trail_raw := builder.client.get("cloudtrail", "get-trail", "Trail", Name=trail_arn):
+                instance = AwsCloudTrail.from_api(trail_raw)
+                builder.add_node(instance, js)
+                collect_status(instance)
+                collect_tags(instance)
+                if instance.trail_has_insight_selectors:
+                    collect_event_selectors(instance)
+                if instance.trail_has_insight_selectors:
+                    collect_insight_selectors(instance)
+
+        def collect_event_selectors(trail: AwsCloudTrail) -> None:
+            trail.trail_event_selectors = []
+            for item in builder.client.list(
+                "cloudtrail", "get-event-selectors", "AdvancedEventSelectors", TrailName=trail.arn
+            ):
+                mapped = bend(AwsCloudTrailEventSelector.mapping, item)
+                trail.trail_event_selectors.append(from_json(mapped, AwsCloudTrailEventSelector))
+
+        def collect_insight_selectors(trail: AwsCloudTrail) -> None:
+            trail.trail_insight_selectors = []
+            for item in builder.client.list(
+                "cloudtrail",
+                "get-insight-selectors",
+                "InsightSelectors",
+                TrailName=trail.arn,
+                expected_errors=["InsightNotEnabledException"],
+            ):
+                trail.trail_insight_selectors.append(item["InsightType"])
+
+        def collect_status(trail: AwsCloudTrail) -> None:
+            status_raw = builder.client.get("cloudtrail", "get-trail-status", Name=trail.arn)
+            mapped = bend(AwsCloudTrailStatus.mapping, status_raw)
+            status = from_json(mapped, AwsCloudTrailStatus)
+            trail.trail_status = status
+            trail.ctime = status.start_logging_time
+            trail.mtime = status.latest_delivery_time
+
+        def collect_tags(trail: AwsCloudTrail) -> None:
+            for tr in builder.client.list("cloudtrail", "list-tags", "ResourceTagList", ResourceIdList=[trail.arn]):
+                trail.tags = bend(S("TagsList", default=[]) >> ToDict(), tr)
+
+        for js in json:
+            if js["HomeRegion"] == builder.region.name:
+                # list trails will return trails for all regions
+                # filter here, to get the assignment to the correct region
+                builder.submit_work(collect_trail, js["TrailARN"])
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if s3 := self.trail_s3_bucket_name:
+            builder.add_edge(self, clazz=AwsS3Bucket, name=s3)
+        if sns := self.trail_sns_topic_arn:
+            builder.add_edge(self, clazz=AwsSnsTopic, arn=sns)
+        if kms := self.trail_kms_key_id:
+            builder.add_edge(self, clazz=AwsKmsKey, id=AwsKmsKey.normalise_id(kms))
```

Comment:
we usually connect Kms Keys as a  dependant node to whatever resource is using it

Reply:
For cloudtrail yu can choose if an existing key should be used or a new one is created.
The key is not dependant on the trail and vice versa.

<img width="320" alt="image" src="https://user-images.githubusercontent.com/330485/216376676-292e3c73-665d-4698-b0b4-6ef7b289b323.png">